### ISSUE TITLE
SSRF what is ifconfig.pro?

### ISSUE THREAD

**Issue by dovecode** [WebGoat/WebGoat#847]
The instruction just says to:

> Change the URL to display the Interface Configuration with ifconfig.pro

Now, call me a newbie, but I had no idea what "ifconfig.pro" was. I assumed it was some sort of misnamed local version of `ifconfig` that was in the image that you wanted me to somehow try to invoke via command injection or something.

How about changing this to:

> Change the URL of the image to display the Interface Configuration using the ifconfig.pro Web site

**Comment by zubcevic** [comment]
Hi and thanks for reviewing all lessons in WebGoat. That is really useful. A lot of work has been done lately to fix a lot of old lessons and to add some new lessons, but you tend to no longer see issues when you work longer on the project. WebGoat has been around for a very long time and a number of lessons are not really maintained by the ones that created them. Also the UI really needs some fixes. I hope you might be interested in making some improvements as well so the things that are unclear get improved. So many things to do and only a few committers

**Comment by dovecode** [comment]
Sure, but to be honest, I am not sure how much of the uncertainty in some of the lessons are intentional and how many are just due to poor communication.

I suppose what I'm lacking is an understanding of a more fundamental principle for lesson structure, guidance, and evaluation. Maybe it is that lack of clarity that has led to some of these frustrating instances.

For example (and apologies for going off topic from this specific issue), I tried to complete the final password reset exercise. I did manage to get the reset link spoofed as required in the password reset mail, but "Tom" (the WebGoat lesson server) never activated the link, or maybe it was WebWolf that never intercepted it? I have no idea if this was caused by a network issue with the Docker setup, or if there was still some missing piece of the puzzle that I didn't get. Without knowing this, it is hard to contribute with a solution...

**Comment by zubcevic** [comment]
I hope you tested it all on the v8.1.0 release.
I ran a test using: docker run --name webgoat -d -p 8080:8080 -p 9090:9090 -e TZ=Europe/Amsterdam webgoat/goatandwolf:v8.1.0 
and the url's http://127.0.0.1/WebGoat and http://127.0.0.1:9090/WebWolf and the password reset lesson works as expected and yes this one is a bit hard.

<img width="595" alt="Schermafbeelding 2020-06-20 om 09 52 54" src="https://user-images.githubusercontent.com/7931455/85196841-426a4a00-b2dd-11ea-866a-cec7f70fa52b.png">

You should first try to send a fake link to yourself, just to see if the changes work correctly and then try to do the same towards tom. Which has a different email domain. Then check the incoming requests on WebWolf

<img width="1290" alt="Schermafbeelding 2020-06-20 om 09 56 06" src="https://user-images.githubusercontent.com/7931455/85196865-76de0600-b2dd-11ea-9117-53145b1562e8.png">

**Comment by zubcevic** [comment]
There is a lesson to teach how to write new lessons in WebGoat

<img width="740" alt="Schermafbeelding 2020-06-20 om 09 54 03" src="https://user-images.githubusercontent.com/7931455/85196882-9c6b0f80-b2dd-11ea-9f3a-263a08e24883.png">

**Comment by dovecode** [comment]
I just tried the password reset one again and it works now. Turns out I made a silly mistake in the format of my link generation which made the new link subtly wrong. I don't want to spoil the solution, but I essentially made "http" the new host, not 127.0.0.1:9090  :)

On the lesson writing guide. What I meant was not the mechanics of creating a lesson, but how to best structure a lesson, how best to teach the concept and provide hints. There's a very broad difference in the quality of the lessons and the presentation, which I can only contribute to the lack of guidance.

By the way, which I don't think the lesson writing guide should be in the app. It should really be a guide on github, not an in-app lesson. Think about it: I've just downloaded and installed WebGoat. I want to do ALL the lessons. These are the lessons I them do:

1. Introduction to WebGoat - great. Now I know what this is
2. Introduction to WebWolf - ok, now I know why there's a WebWolf as well!
3. HTTP Basics - erhm... OK. Should this by required background -- seems a bit basic
4. HTTP Proxies - This spends most of the time telling me how to set up ZAP, which is great and all. Right up to the last step where it tells me that Burp is probably better, as "one of the exercises" can't be done with ZAP
5. Developer Tools - this should probably come before the proxy setup
6. CIA - fine introduction
7. Nice, except there's no suggestion of any of the tools to use here. If I already know how to use openssl and the like to generate keys, etc, the content is pointless, but if I don't, solving the exercises is very hard.
8. Writing a new lesson - erhm... what? I'm here to learn to exploit vulnerabilities. Why is this a lesson? Is this somehow important to understancing Web vulnerabilities?


This actually points to a bit of a wider problem with WebGoat, which I don't know if is solvable: The barrier between WebGoat-the-lesson-app and WebGoat-the-vulnerable-app is very blurry. This also causes some weird moments where (as with the password reset above), I have to pass in a session ID to reset my password. Essentially, I cannot reset my password without logging in. I know these are two different authentications (one to WebGoat-the-lesson-app and the other "authenticating" in WebGoat-the-vulnerable-app).
